scanner/src/test/kotlin/scanners/fossid/FossIdTest.kt

```diff
@@ -103,6 +103,18 @@ class FossIdTest : WordSpec({
         }
     }
 
+    "convertGitUrlToProjectName()" should {
+        "extract the repository name from the git URL without the .git suffix" {
+            val httpGitUrl = FossId.convertGitUrlToProjectName("http://github.com/path/to/example-repo.git")
+            val httpsGitUrl = FossId.convertGitUrlToProjectName("https://github.com/path/to/example-repo.git")
+            val httpsGitUrlDottedRepoName = FossId.convertGitUrlToProjectName("https://github.com/example.repo.git")
```

Comment:
I believe we need different test cases here. Instead of only using GitHub as a host, please use the GitHub URL mentioned in the function docs for `convertGitUrlToProjectName` ("https://github.com/jshttp/mime-types.git"), and additionally a URL to a Git repo hosted on Azure, like "https://dev.azure.com/org/project/_git/repo".

@nnobelis do you have more URLs / hosts in mind that should be tested?

Reply:
These are the two main ones. Maybe one without `-` such as `https://github.com/vdurmont/semver4j.git`

BTW this is `https://dev.azure.com/example/_git/project` so the project name is what is coming after the `_git`.

Reply:
> BTW this is `https://dev.azure.com/example/_git/project` so the project name is what is coming after the `_git`.

Well, we're using the (Azure) repo name as the (ORT) project name. The Azure project name, however, indeed is the part *before* "_git".